pyproject.toml

```diff
@@ -0,0 +1,94 @@
+[project]
+name = "HOTOSM Tasking Manager"
+version = "0.1.0"
+description = "The tool to team up for mapping in OpenStreetMap"
+authors = [
+    {name = "HOT Sysadmin", email = "sysadmin@hotosm.org"},
+]
+dependencies = [
+    "alembic==1.4.3",
+    "aniso8601==8.0.0",
+    "appdirs==1.4.4",
+    "APScheduler==3.7.0",
+    "attrs==20.2.0",
+    "black==20.8b1",
+    "bleach==5.0.1",
+    "blinker==1.5",
+    "cachetools==4.1.1",
+    "certifi==2020.6.20",
+    "chardet==3.0.4",
+    "charset-normalizer==2.0.12",
+    "click==7.1.2",
+    "coverage==6.4",
+    "entrypoints==0.3",
+    "flake8==3.8.4",
+    "Flask==1.1.2",
+    "Flask-Cors==3.0.9",
+    "Flask-HTTPAuth==4.1.0",
+    "Flask-Mail==0.9.1",
+    "Flask-Migrate==2.5.3",
+    "Flask-RESTful==0.3.8",
+    "Flask-Script==2.0.6",
+    "Flask-SQLAlchemy==2.4.4",
+    "flask-swagger==0.2.14",
+    "GeoAlchemy2==0.8.4",
+    "geojson==1.3.4",
+    "gevent==21.12.0",
+    "glob2==0.7",
+    "greenlet==1.1.2",
+    "gunicorn==20.0.4",
+    "idna==2.10",
+    "importlib-metadata==4.13.0",
+    "itsdangerous==1.1.0",
+    "Jinja2==2.11.3",
+    "Mako==1.2.2",
+    "Markdown==3.3.3",
+    "MarkupSafe==1.1.1",
+    "mccabe==0.6.1",
+    "mypy-extensions==1.0.0",
+    "newrelic==8.8.0",
+    "nose==1.3.7",
+    "oauthlib==3.2.0",
+    "pathspec==0.11.1",
+    "psycopg2==2.8.6",
```

Comment:
'pdm install' failed to install psycopg2 while setting up without docker. However, everything works fine with the docker-setup. Is this a problem with my system?

Reply:
@Aadesh-Baral what's the error message you get?

Reply:
ERRORS:
add psycopg2 failed:
Traceback (most recent call last):
  File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/aadesh/.local/lib/python3.8/site-packages/pdm/installers/synchronizers.py", line 236, in install_candidate
    self.manager.install(can)
  File "/home/aadesh/.local/lib/python3.8/site-packages/pdm/installers/manager.py", line 39, in install
    installer(str(prepared.build()), self.environment, prepared.direct_url())
  File "/home/aadesh/.local/lib/python3.8/site-packages/pdm/models/candidates.py", line 375, in build
    builder.build(build_dir, metadata_directory=self._metadata_dir)
  File "/home/aadesh/.local/lib/python3.8/site-packages/pdm/builders/wheel.py", line 30, in build
    filename = self._hook.build_wheel(out_dir, config_settings, metadata_directory)
  File "/home/aadesh/.local/lib/python3.8/site-packages/pyproject_hooks/_impl.py", line 209, in build_wheel
    return self._call_hook('build_wheel', {
  File "/home/aadesh/.local/lib/python3.8/site-packages/pyproject_hooks/_impl.py", line 311, in _call_hook
    self._subprocess_runner(
  File "/home/aadesh/.local/lib/python3.8/site-packages/pdm/builders/base.py", line 261, in subprocess_runner
    return log_subprocessor(cmd, cwd, extra_environ=env)
  File "/home/aadesh/.local/lib/python3.8/site-packages/pdm/builders/base.py", line 107, in log_subprocessor
    raise build_error(e) from None
pdm.exceptions.BuildError: Build backend raised error: Showing the last 10 lines of the build output:
It appears you are missing some prerequisite to build the package from source.

You may install a binary package by installing 'psycopg2-binary' from PyPI.
If you want to install psycopg2 from source, please install the packages
required for the build and try again.

For further information please check the 'doc/src/install.rst' file (also at
<https://www.psycopg.org/docs/install.html>).

error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1

Reply:
You might have to install `libpq-dev` in order to install from source. Can you try that?

Reply:
Installing psycopg2-binary also does not resolve the problem.

Reply:
`libpq-dev `already exists in my system.

Reply:
The Dockerfile uses Apline and install `postgresql-dev`, plus is running Python 3.10.

Looks like you are running Python 3.8, but from what distro of Linux?

I'm assuming Ubuntu, or Debian here.
The version of psycopg2 being installed is 2.8.6: https://packages.debian.org/bullseye/python3-psycopg2
So you will require libpq-dev v13.10: https://packages.debian.org/bullseye/libpq-dev

I would recommend attempting to replicate the environment of the docker build.
Run a `debian:bullseye` image and try to install again.

```bash
# Start with a fresh environment
docker run --rm -it -v $PWD:$PWD debian:bullseye bash
# apt install libpq-dev
apt update && apt install -y libpq-dev
# cd to your code
pip install pdm
pdm install
```

Hope this helps.

Reply:
Looks like installing psycopg2-binary==2.9.6 instead of psycopg2==2.8.6 will resolve this issue.

Reply:
I would advise against that & instead install the correct prerequisite libraries on your system 👍

 https://stackoverflow.com/questions/70330567/what-is-the-different-about-psycopg2-and-psycopg2-binary-python-package